### PR TITLE
Dynamic routes clarification

### PR DESCRIPTION
**Pull request by sgmonda**
As talked in https://github.com/zeit/next.js/issues/8101 dynamic routing documentation could be confusing. This PR tries to clarify it a bit.

### CODE REVIEW

File: packages/next/README.md [hunk 1]
```diff
@@ -311,8 +311,20 @@ A `<Link>` for `/post/abc` looks like so:
 </Link>
 ```
 
-- `href`: the path inside `pages` directory
-- `as`: the path that will be rendered in the browser URL bar
+- `href`: the path inside `pages` directory.
+- `as`: the path that will be rendered in the browser URL bar.
+
+As `href` is a filesystem path, it shouldn't change during execution, while you will probably need to change `as`
```
Comment (Timer) [inline packages/next/README.md]:
```suggestion
As `href` is a filesystem path, it shouldn't change at runtime, instead, you will probably need to change `as`
```

File: packages/next/README.md [hunk 2]
```diff
@@ -568,8 +580,8 @@ Example:
    <Link href="/post?slug=something" as="/post/something">
    ```
 
-   - `href`: the path inside `pages` directory
-   - `as`: the path used by your server routes
+   - `href`: the path inside `pages` directory. Only right side of `?` character could change dynamically during execution, if needed, as left side is a filesystem path.
```
Comment (Timer) [inline packages/next/README.md]: Let's leave this one as-is since it's not related to dynamic routing.

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/next/README.md (hunk 2 above)
Comment (sgmonda) [reply packages/next/README.md]: I agree. Changes removed in https://github.com/zeit/next.js/pull/8110/commits/d217c142acb3a587a555497ad7443ecc12c55367

**Review by Timer — APPROVED** [review]

**Comment by timneutkens** [comment]
Great update @sgmonda!

**Pull request merged**